src/include/container/hash/extendible_hash_table.h

```diff
@@ -172,6 +172,12 @@ class ExtendibleHashTable : public HashTable<K, V> {
   mutable std::mutex latch_;
   std::vector<std::shared_ptr<Bucket>> dir_;  // The directory of the hash table
 
+  /**
+   * @brief Redistribute the kv pairs in a full bucket.
+   * @param bucket The bucket to be redistributed.
+   */
+  auto RedistributeBucket(std::shared_ptr<Bucket> bucket) -> void;
```

Comment:
This is a private helper method for the ref solution so I don't think we should push it to public. Is it not working well with hacky_sync?

Reply:
I put it here with intention. 🤣 As we didn't give the implementation of this function, I think it would be fine.

The bad side of ignoring this file from synchronizing between two repos are that, if you make any change in the public repo, you will need to carefully merge it with private repo, and it will go out-of-sync very soon. There're a lot of examples in this PR, like we have `const` in private repo but not in public, etc. So I think it would be better to have a smaller set of ignored files.

Reply:
We probably want to put this file in the .syncignore file to not sync it? And only expose the public apis of extendible hash table in the public repo

Reply:
Make sense! Let's sync it then haha

Reply:
> Also this hints the students "you might want to implement this function to make your life easier". If they don't want it, they can remove it.

Can you please add this as a comment above the function, for any such function? Even something like `// Optional, feel free to delete.`

Reply:
It's already in the beginning of the private section:

```
 private:
  // TODO(student): You may add additional private members and helper functions
```

I'll make it more detailed 🤤

Reply:
Change LGTM, thanks! Now it should be very clear